src/Microsoft.DotNet.ProjectModel.Server/ProjectStateResolver.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.ProjectModel.Compilation;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.DotNet.ProjectModel.Server.Helpers;
+using Microsoft.DotNet.ProjectModel.Server.InternalModels;
+using Microsoft.DotNet.ProjectModel.Server.Models;
+
+namespace Microsoft.DotNet.ProjectModel.Server
+{
+    internal class ProjectStateResolver
```

Comment:
It's not that simple. Through `ProjectContext` represents most of the data to diff, it still needs a scan of the diagnostics and libraries to build up the actual data model for returning data. For example: additional diagnostics are created when a dependencies flip from project to package. 

Some of the data transform can be delayed to the point when actual response is determined to be required. I'll move them to independent logic.